src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/ObjectWriter.cs

```diff
@@ -1150,14 +1150,26 @@ public static void EmitObject(string objectFilePath, IReadOnlyCollection<Depende
                 }
 
                 if (logger.IsVerbose)
-                    logger.LogMessage($"Finalizing output to '{objectFilePath}'...");
+                    logger.LogMessage($"Emitting debug information");
+
+                // Native side of the object writer is going to do more native memory allocations.
+                // Free up as much memory as possible so that we don't get OOM killed.
+                // This is potentially a waste of time. We're about to end the process and let the
+                // OS "garbage collect" the entire address space.
+                // https://github.com/dotnet/runtime/issues/83180
+                GC.Collect();
```

Comment:
We may want to do this only when the process is using large portion of total available memory. Something like:
```
var gcMemoryInfo = GC.GetGCMemoryInfo();
if (gcMemoryInfo.TotalCommittedBytes > gcMemoryInfo.TotalAvailableMemoryBytes / 2)
```

Reply:
Is this a better heuristic than `AddMemoryPressure`? I filed #83180 to figure out the right way to do this. So just wondering whether to use your heuristic and close the issue, or don't try to be fancy and just do something to unblock https://github.com/dotnet/performance/pull/2919.

Reply:
Or I guess it doesn't hurt to just add this...

Reply:
AddMemoryPressure only works for cases that depend on finalizers for cleanup (e.g. COM interop). It does not help when you use proper Dispose pattern for cleanup.

AddMemoryPressure is not actually part of the core GC. It is heuristics that is independent on the GC. It keeps track of how much pressure you have added and calls `GC.Collect` when you have added enough.

Reply:
I do not think there are better ways to do this. The exact threshold when to trigger the GC may be adjusted up or down, but it is not an exact science. 50% of available memory sounds about right to me.